Homework_Basic_Python.ipynb

```diff
@@ -0,0 +1,209 @@
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Length of array: 4\n",
+      "enter number 4\n",
+      "enter number 4\n",
+      "enter number 4\n",
+      "enter number 4\n",
+      "Sum of numbers is this array is 16\n"
+     ]
+    }
+   ],
+   "source": [
+    "dos = []\n",
+    "num = int(input('Length of array: '))\n",
+    "for n in range(num):\n",
+    "        numbers=(int(input('enter number ')))\n",
+    "        dos.append(numbers)\n",
+    "print('Sum of numbers is this array is', sum(dos))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Assignment 3\n",
+    "\n",
+    "Adding to assignment 2, implement the same functionality but allow the array to be a mix of numbers and numbers represented as strings, e.g. [1, 5, '7', 5, '10']."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Length of array: 3\n",
+      "enter number or string 3\n",
+      "enter number or string '4'\n",
+      "enter number or string 4\n",
+      "['3', \"'4'\", '4']\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "tres=[]\n",
+    "num = int(input('Length of array: '))\n",
+    "for n in range(num):\n",
+    "        numbers=(input('enter number or string '))\n",
+    "        tres.append(numbers)\n",
+    "print(tres)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Assignment 4\n",
+    "\n",
+    "Let's build a small database program. The goal is to let the user build a database of pets. The database is a dictionary, in which a pet's name acts as the key and the value holds the type of animal (e.g. 'kitty': 'cat'). We'll assume that no two animals have the same name. Use a `while` loop to allow a user to add entries by first asking for the name of the pet and then for its type. Then ask the user if they want to add another entry.\n",
+    "    \n",
+    "*Tip*: the while loop condition can be a statement such as variables equals or does not equal a value."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "SyntaxError",
+     "evalue": "invalid syntax (<ipython-input-29-0dc388d5e7be>, line 6)",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-29-0dc388d5e7be>\"\u001b[0;36m, line \u001b[0;32m6\u001b[0m\n\u001b[0;31m    reply = input('Type 'yes' to add another pet: ')\u001b[0m\n\u001b[0m                           ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
+     ]
+    }
+   ],
+   "source": [
+    "while True:\n",
+    "    name = input('animal name?: ')\n",
+    "    type = input('animal type?: ')\n",
+    "    pets = {}\n",
```

Comment:
by doing this inside the loop, you're resetting the dictionary to an empty dictionary on every loop, so it will never contain more than one entry.